src/mantine-dates/src/components/PickerInputBase/PickerInputBase.tsx

```diff
@@ -155,12 +157,12 @@ export const PickerInputBase = forwardRef<HTMLButtonElement, PickerInputBaseProp
           opened={dropdownOpened}
           onClose={handleClose}
           disabled={dropdownType === 'modal' || readOnly}
-          trapFocus
+          trapFocus={!hasLabel}
```

Comment:
Why focus is trapped only when the input has label?

Reply:
Actually, it's the other way around. We don't want to focus to be trapped when we have a label.

So if we will have `trapFocus: true` by default focus will switch instantly from the input field and `Popover.Target` will try to trap focus on the calendar element. (You can see a small glitch of blue outline around input for a moment). 
 
By adding `trapFocus={!hasLabel}` we are trying to disable that behaviour and keep the focus on the input field. If there is no label provided to the input field then it's okay to trap the focus on the calendar.
 
Here is the recording:
https://github.com/mantinedev/mantine/assets/38986496/c55c9705-9e84-4a75-9128-a58e2d41123e

Reply:
Focus trap is the correct behavior in both cases – if it is not trapped, then keyboard controls do not work correctly and thus UX suffers.

Reply:
What do you mean by keyboard controls do not work correctly? If we don't have a focus trap we will be focused on the input field and if the user will press `Tab` to navigate to the next interactive element it will be the direction buttons of `Calendar`.

```jsx
trapFocus={!hasLabel}
// trapFocus
```

![image](https://github.com/mantinedev/mantine/assets/38986496/627fcb29-4583-4b5d-9beb-0f4a103c0f55)

```jsx
// trapFocus={!hasLabel}
trapFocus
```
![image](https://github.com/mantinedev/mantine/assets/38986496/20feed8f-21b8-4e94-a1a1-d85aa9799f80)

Reply:
- you can still escape the dropdown when pressing tab key
- when the input is clicked focus is not shifted to the dropdown – users with screen readers will be confused, they've performed an action and there was no announced result

Disabling focus trap is accessibility violation

Reply:
Sorry very busy couple of weeks, couldn't get back to you. Yeah, I guess you are right, I pushed a change that reverts `trapFocus`.